doc/source/data/dataset-internals.rst

```diff
@@ -114,9 +114,11 @@ This section describes how Datasets manages execution and object store memory.
 Execution Memory
 ~~~~~~~~~~~~~~~~
 
-During execution, certain types of intermediate data must fit in memory. This includes the input block of a task, as well as at least one of the output blocks of the task (when a task has multiple output blocks, only one needs to fit in memory at any given time). The input block consumes object stored shared memory (and Python heap memory if conversion to non-Arrow format is needed). The output blocks consume Python heap memory (prior to putting in the object store) as well as object store memory (after being put in the object store).
+During execution, a task can read multiple input blocks, and write multiple output blocks. The input blocks consume object stored shared memory (and Python heap memory if conversion to non-Arrow format is needed). The output blocks consume Python heap memory (prior to putting in the object store) as well as object store memory (after being put in the object store).
```

Comment:
```suggestion
During execution, a task can read multiple input blocks, and write multiple output blocks. Input and output blocks consume both worker heap memory and shared memory via Ray's object store.
```

Reply:
This is simpler and I don't think the other details are necessary.

Reply:
@stephanie-wang - thanks, changed.